datasets/web_nlg/web_nlg.py

```diff
@@ -0,0 +1,247 @@
+        ),
+        datasets.BuilderConfig(
+            name="release_v3.0_ru", description="WebNLG+ data used in the WebNLG challenge 2020. Russian."
+        ),
+    ]
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "category": datasets.Value("string"),
+                "size": datasets.Value("int32"),
+                "eid": datasets.Value("string"),
+                "original_triple_sets": datasets.Sequence(
+                    datasets.Features({"otriple_set": datasets.Sequence(datasets.Value("string"))}),
+                ),
+                "modified_triple_sets": datasets.Sequence(
+                    datasets.Features({"mtriple_set": datasets.Sequence(datasets.Value("string"))}),
+                ),
+                "shape": datasets.Value("string"),
+                "shape_type": datasets.Value("string"),
+                "lex": datasets.Sequence(
+                    datasets.Features(
+                        {
+                            "comment": datasets.Value("string"),
+                            "lid": datasets.Value("string"),
+                            "text": datasets.Value("string"),
+                        }
+                    ),
+                ),
+                "2017_test_category": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage="https://webnlg-challenge.loria.fr/",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=spl,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filedirs": [
+                        os.path.join(data_dir, "webnlg-dataset-master", dir_suf) for dir_suf in dir_suffix_list
+                    ],
+                },
+            )
+            for spl, dir_suffix_list in _FILE_PATHS[self.config.name].items()
+        ]
+
+    def _generate_examples(self, filedirs):
+        """ Yields examples. """
+
+        id_ = 0
+        for xml_location in filedirs:
+            for xml_file in glob(pjoin(xml_location, "*.xml")):
```

Comment:
Could you sort the files using `sorted` for examples ?
Since we want the dataset to always have the examples in the same order, then we want the script to always iter through file list in the same order, no matter the environment.